tests/test_Canvas.py

```diff
@@ -0,0 +1,55 @@
+"""This module contains unit tests for the Catalog class"""
+
+__author__ = "Siavash Yasini"
+__email__ = "yasini@usc.edu"
+
+import pytest
+
+
+def test_clean_canvas(test_canvas):
+    """Passes if after clean.canvas()
+    the status of pixels, alms, and Cls are correctly updated """
+    test_canvas.clean()
+    assert not test_canvas._pixel_is_outdated
+    assert test_canvas._alm_is_outdated
+    assert test_canvas._Cl_is_outdated
```

Comment:
Is there no other, more obvious, way to see if the canvas has been cleaned?  It's nice if tests can provide a way for a reader to understand your code (and "your reader" might be you in 3 months).  When I look at this, these assertions aren't obviously tests of a clean canvas.  I kind of expected to see something like (and I haven't looked at the rest of the code, so this might be way off base!)
```python
assert (test_canvas.pixels == 0).all(), f"Pixels not all zero: {test_canvas.pixels}"
```

Also, I recommend having a descriptive string for if the test fails.

Reply:
Aah! I see that I haven't been careful enough with the name that I chose for this test. 

This is testing the behavior of these private attributes upon using canvas.clean() and it's not testing if the canvas is actually clean! I should also add the test that you suggested here to see if `canvas.clean()` is doing its job. But here I'm just making sure that when the canvas is cleaned, "alm" and "Cl" will become outdated automatically (This will ensure that when their properties (`.alm`, and `.Cl`) are called, they are recalculated). 

The reason I wrote this test is that the "`_is_outdated`" status of pixels, alms, and Cls are important for keeping them in sync (this is implemented in the code). So the idea is to check whether for example setting all map pixels to zero will also make Cl outdated or not.

Do you have any suggestions for what I should call this instead? Based on your previous comment I would probably go with `test_alm_Cl_outdated_upon_canvas_clean()` or something like that?

Reply:
Makes sense, and yes, that sounds good to me.  Within reason, always err on the side of verbosity/clarity.

Reply:
@dsspiegel I added two more commits 👇implementing the changes suggested above.